src/pykeen/utils.py

```diff
@@ -1630,3 +1632,56 @@ def determine_maximum_batch_size(batch_size: int | None, device: torch.device, m
             )
         logger.debug(f"Automatically set maximum batch size to {batch_size=:_}")
     return batch_size
+
+
+F = TypeVar("F", bound=Callable)
+
+
+def add_doc_note_about_resolvers(
+    *params: str,
+    resolver_name: str,
+) -> Callable[[F], F]:
+    """
+    Build a decorator to add information about resolved parameter pairs.
+
+    :param params:
+        the name of the parameterss. Will be automatically completed to include all the ``_kwargs`` suffixed parts, too.
+    :param resolver_name:
+        the fully qualified path of the resolver used to construct a reference via the ``:data:`` role.
+
+    :return:
+        a decorator for function.
+    """
+
+    def add_note(func: F) -> F:
+        """
+        Extend the function's docstring with a note about resolved parameters.
+
+        :param func:
+            the function to decorate.
+
+        :return:
+            the function with extended docstring.
+
+        :raises ValueError:
+            when the signature does not contain the resolved parameter names.
+        """
+        signature = inspect.signature(func)
+        if missing := set(f"{p}{suffix}" for p in params for suffix in ("", "_kwargs")).difference(
+            signature.parameters
+        ):
+            raise ValueError(f"{missing=} parameters in {signature=}.")
+        old_doc = func.__doc__
+        assert old_doc is not None
+        pairs_str = ", ".join(f"``({param}, {param}_kwargs)``" for param in params)
+        func.__doc__ = textwrap.dedent(
+            f"""
+            {old_doc}
+
+            .. note ::
```

Comment:
This seems to somehow break the indentation.

Reply:
![Screenshot 2024-09-13 214354](https://github.com/user-attachments/assets/5826a106-6249-4af7-b0c3-441d4e3473c2)

Reply:
I fixed this

Reply:
actually, calling it fixed is a bit much. this borks all of the other styling

Reply:
maybe we should change the decorator to operate on the class level documentation, and to look inside the `__init__`?

Reply:
I was quite unclear to me how and when those are composed 😅

Reply:
Now in
```python
from pykeen.nn.modules import CrossEInteraction

help(CrossEInteraction)
```
the docstring looks fine 🤔

Reply:
And now (after [bring back lstrip](https://github.com/pykeen/pykeen/pull/1447/commits/c7dbc5070c0d5f626fa7317f36e19cf3de7dd827)) it looks fine 🎉

![Screenshot 2024-09-14 120624](https://github.com/user-attachments/assets/b12d997b-477f-4d48-84fb-486ebbfe83ec)